evaluate/task/task-write-test_test.go

```diff
@@ -216,3 +217,114 @@ func TestTaskWriteTestsRun(t *testing.T) {
 		})
 	})
 }
+
+func TestValidateWriteTestsRepository(t *testing.T) {
+	type testCase struct {
+		Name string
+
+		Before func(repositoryPath string)
+
+		RepositoryPath string
+		Language       language.Language
+
+		ExpectedErr string
+	}
+
+	validate := func(t *testing.T, tc *testCase) {
+		t.Run(tc.Name, func(t *testing.T) {
+			logOutput, logger := log.Buffer()
+			defer func() {
+				if t.Failed() {
+					t.Logf("Logging output: %s", logOutput.String())
+				}
+			}()
+
+			temporaryDirectoryPath := t.TempDir()
+			repositoryPath := filepath.Join(temporaryDirectoryPath, "testdata", tc.RepositoryPath)
+			require.NoError(t, osutil.CopyTree(filepath.Join("..", "..", "testdata", tc.RepositoryPath), repositoryPath))
+
+			if tc.Before != nil {
+				tc.Before(repositoryPath)
+			}
+
+			actualErr := validateWriteTestsRepository(logger, repositoryPath, tc.Language)
+			if len(tc.ExpectedErr) > 0 {
+				require.ErrorContains(t, actualErr, tc.ExpectedErr)
+			} else {
+				require.NoError(t, actualErr)
+			}
+		})
```

Comment:
I think at this point it would be cool to refactor this validation (and the `testCase`) out and reuse across the different tasks.